Modules/Quotas/src/Quotas.Domain/Aggregates/Identities/Quota.cs

```diff
@@ -14,9 +16,29 @@ protected Quota(string applyTo)
 
     public QuotaId Id { get; }
     public string ApplyTo { get; }
-    public DateTime? IsExhaustedUntil { get; }
+    public DateTime? IsExhaustedUntil { get; private set; }
     public abstract int Weight { get; }
     public abstract MetricKey MetricKey { get; }
     public abstract int Max { get; }
     public abstract QuotaPeriod Period { get; }
+    public DateTime PeriodEndTime => Period.CalculateEnd();
+    public DateTime PeriodBeginTime => Period.CalculateBegin();
+
+    // TODO: What is the definition of a Valid quota?
+    public bool IsCurrentlyValid()
```

Comment:
This method shouldn't currently be needed. Only when we have the `ValidFrom` and `ValidTo` properties